solo-chains/runtime/starlight/src/tests/relay_registrar.rs

```diff
@@ -351,6 +350,13 @@ fn deregister_calls_schedule_para_cleanup() {
                 Configuration::set_max_head_data_size(root_origin(), 20500),
                 ()
             );
+
+            // Call end_block() to ensure that parachains_shared::CurrentSessionIndex
+            // storage gets updated properly inside on_finalize() and
+            // matches the one inside pallet_session::CurrentIndex.
+            end_block();
+            start_block();
```

Comment:
`run_to_session()` is doing exactly this internally so you can remove all the calls you added as they won't affect the test. If you need to produce an extra block somewhere, use `run_block()`, but you probably don't need it.

Reply:
Yes only need to use `end_block()` after the last `run_to_session()`. The previous behavior was needed when we had the call to `InnerRegistrar::deregister` inside `on_initialize()`, but now that we have everything inside `on_finalize()` we don't need it anymore.

Reply:
_fixed_